packages/gestalt/src/Typeahead.js

```diff
@@ -0,0 +1,341 @@
+
+      handleBlur({ event });
+    }
+    // Scrolling
+    handleScrolling(direction);
+  };
+
+  const handleHover = (ref, index) => {
+    setHoveredItem(index);
+  };
+
+  return (
+    <Box ref={forwardedRef}>
+      <TypeaheadInputField
+        label={label}
+        id={id}
+        value={search}
+        placeholder={placeholder}
+        size={size}
+        onChange={handleChange}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        onClear={handleClear}
+        onKeyNavigation={handleKeyNavigation}
+        setContainer={setContainerOpen}
+        ref={inputRef}
+      />
+
+      {containerOpen && inputRef.current && (
+        <Layer>
+          <Flyout
+            showCaret={false}
+            anchor={inputRef.current}
+            idealDirection="down"
+            onDismiss={() => {}}
+            positionRelativeToAnchor={false}
+            size="flexible"
+            // Forces the flyout to re-render and adjust it's position correctly
+            key={options.length}
+          >
+            <Box
+              ref={containerRef}
+              position="relative"
+              overflow="auto"
+              padding={1}
+              marginTop={2}
+              marginBottom={2}
+              maxHeight="50vh"
+              width={inputRef?.current?.offsetWidth}
+              color="white"
+            >
+              <Box alignItems="center" direction="column" display="flex">
+                {/* Handle when no results */}
+                {options.length === 0 && (
+                  <Box margin={2}>
+                    <Text color="gray">{noResultText}</Text>
+                  </Box>
+                )}
+                {/* Return options */}
+                {options.map((option, index) => (
+                  <Option
+                    index={index}
+                    key={`${option[searchField] + index}`}
+                    option={option}
+                    searchField={searchField}
+                    selected={selected}
+                    hoveredItem={hoveredItem}
+                    setHoveredItem={handleHover}
+                    handleSelect={handleSelect}
+                    setOptionRef={setOptionRef}
+                  />
+                ))}
+              </Box>
+            </Box>
+          </Flyout>
+        </Layer>
+      )}
+    </Box>
+  );
+};
+
+Typeahead.displayName = 'Typeahead';
+
+Typeahead.propTypes = {
+  id: PropTypes.string,
+  onChange: PropTypes.func,
+  data: PropTypes.arrayOf(
+    PropTypes.exact({
+      label: PropTypes.string.isRequired,
+      value: PropTypes.string.isRequired,
+    })
+  ),
+  placeholder: PropTypes.string,
+  size: PropTypes.oneOf(['md', 'lg']),
+  searchField: PropTypes.string,
+  onBlur: PropTypes.func,
+  onFocus: PropTypes.func,
+  onSelect: PropTypes.func,
+  defaultItem: PropTypes.exact({
+    label: PropTypes.string.isRequired,
+    value: PropTypes.string.isRequired,
+  }),
+  noResultText: PropTypes.string,
```

Comment:
@simkessy 
* `label` seems to be missing from this list
* `data` / `id` and `noResultText` should all be required